### PR TITLE
Updated iPhone enable developer mode instructions

### PR DESCRIPTION
**Pull request by mavwolverine**
The instructions specify to enable Developer Mode in Privacy and Security. But Developer Mode option is not available, unless you open Xcode and reconnect the device.

___________
9/23 edit from @andrewkolos: See https://stackoverflow.com/questions/77648341/flutter-not-reading-device-developer-mode-setting

## Pre-launch Checklist

- [x] I read the [Contributor Guide] and followed the process outlined there for submitting PRs.
- [x] I read the [Tree Hygiene] wiki page, which explains my responsibilities.
- [x] I read and followed the [Flutter Style Guide], including [Features we expect every widget to implement].
- [x] I signed the [CLA].
- [ ] I listed at least one issue that this PR fixes in the description above.
- [x] I updated/added relevant documentation (doc comments with `///`).
- [ ] I added new tests to check the change I am making, or this PR is [test-exempt].
- [ ] I followed the [breaking change policy] and added [Data Driven Fixes] where supported.
- [ ] All existing and new tests are passing.

If you need help, consider asking for advice on the #hackers-new channel on [Discord].

[Contributor Guide]: https://github.com/flutter/flutter/blob/main/docs/contributing/Tree-hygiene.md#overview
[Tree Hygiene]: https://github.com/flutter/flutter/blob/main/docs/contributing/Tree-hygiene.md
[test-exempt]: https://github.com/flutter/flutter/blob/main/docs/contributing/Tree-hygiene.md#tests
[Flutter Style Guide]: https://github.com/flutter/flutter/blob/main/docs/contributing/Style-guide-for-Flutter-repo.md
[Features we expect every widget to implement]: https://github.com/flutter/flutter/blob/main/docs/contributing/Style-guide-for-Flutter-repo.md#features-we-expect-every-widget-to-implement
[CLA]: https://cla.developers.google.com/
[flutter/tests]: https://github.com/flutter/tests
[breaking change policy]: https://github.com/flutter/flutter/blob/main/docs/contributing/Tree-hygiene.md#handling-breaking-changes
[Discord]: https://github.com/flutter/flutter/blob/main/docs/contributing/Chat.md
[Data Driven Fixes]: https://github.com/flutter/flutter/blob/main/docs/contributing/Data-driven-Fixes.md

### CODE REVIEW

**Comment by andrewkolos** [comment]
> The instructions specify to enable Developer Mode in Privacy and Security. But Developer Mode option is not available, unless you open Xcode and reconnect the device.

@jmagman Have you ever heard of this before? That is, having to open xCode and disconnect/reconnect your device before being able to enable Developer Mode on the device? I'm questioning whether or not there is actually a tool bug here.

**Comment by virajkanwade** [comment]
> > The instructions specify to enable Developer Mode in Privacy and Security. But Developer Mode option is not available, unless you open Xcode and reconnect the device.
> 
> @jmagman Have you ever heard of this before? That is, having to open Xcode and disconnect/reconnect your device before being able to enable Developer Mode on the device? I'm questioning whether or not there is actually a tool bug here.

That is what worked for me, so I added the PR. See also
https://stackoverflow.com/questions/77648341/flutter-not-reading-device-developer-mode-setting

File: packages/flutter_tools/lib/src/runner/target_devices.dart [hunk 1]
```diff
@@ -29,7 +29,7 @@ String _foundSpecifiedDevicesMessage(int count, String deviceId) =>
 String _noMatchingDeviceMessage(String deviceId) => 'No supported devices found with name or id '
     "matching '$deviceId'.";
 String flutterSpecifiedDeviceDevModeDisabled(String deviceName) => 'To use '
-    "'$deviceName' for development, enable Developer Mode in Settings → Privacy & Security.";
+    "'$deviceName' for development, open Xcode, reconnect the '$deviceName' and enable Developer Mode in Settings → Privacy & Security.";
```
Comment (andrewkolos) [inline packages/flutter_tools/lib/src/runner/target_devices.dart:32]: > Also see
> https://stackoverflow.com/questions/77648341/flutter-not-reading-device-developer-mode-setting

Thanks for the link! I can see now that this is a legitimate UX problem.

I think the text change proposed here is a little confusing (though it's possible this is due to me not testing on physical iOS devices very often). When I first read it, I thought it was telling me to change settings in Xcode, not on the device.

Referencing the StackOverflow thread linked, I propose this as a starting point:

```
To use <device> for development, enable Developer Mode in Settings → Privacy & Security on your device.

If this does not work, open Xcode, reconnect the device, and look for a popup on the device asking you to trust this computer.
```

Comment (jmagman) [reply packages/flutter_tools/lib/src/runner/target_devices.dart:32]: @andrewkolos I agree, your wording is better.

Comment (andrewkolos) [reply packages/flutter_tools/lib/src/runner/target_devices.dart]: Updated in commit [reword message](https://github.com/flutter/flutter/pull/154511/commits/99c01adee6d94a43b83f38b6e5f530ca621daac1)

**Comment by andrewkolos** [comment]
Unlisting myself as a reviewer since I have directly pushed changes (context: https://github.com/flutter/flutter/pull/154511#discussion_r1795976772)

**Review by jmagman — APPROVED** [review]

**Comment by jmagman** [comment]
> Unlisting myself as a reviewer since I have directly pushed changes (context: [#154511 (comment)](https://github.com/flutter/flutter/pull/154511#discussion_r1795976772))

@andrewkolos I hereby bless you to give you the second LGTM 🙏

@virajkanwade Thank you for the contribution!

**Review by andrewkolos — APPROVED** [review]

**Pull request merged**